src/Storage/Bucket.php

```diff
@@ -836,4 +853,19 @@ private function isObjectNameRequired($data)
 
         return false;
     }
+
+    /**
+     * Add the billing project ID in to a request.
+     *
+     * @param array $args
+     * @return array
+     */
+    public function addUserProject(array $args)
```

Comment:
WDYT about adding `userProject` to `$this->identity`? Since every operation will require the `userProject` anyways, it will have the added benefit of not needing run this computation on every call .

Also, as a side note: does this need to be public?